dipy/segment/bundles.py

```diff
@@ -0,0 +1,406 @@
+from dipy.tracking.distances import (bundles_distances_mdf,
+                                     bundles_distances_mam)
+from dipy.align.streamlinear import (StreamlineLinearRegistration,
+                                     BundleMinDistanceMetric,
+                                     BundleSumDistanceMatrixMetric,
+                                     BundleMinDistanceAsymmetricMetric)
+from dipy.align.bundlemin import distance_matrix_mdf
+from time import time
+from itertools import chain
+
+from dipy.tracking.streamline import Streamlines
+from nibabel.affines import apply_affine
+
+
+class RecoBundles(object):
+
+    def __init__(self, streamlines, cluster_map=None, clust_thr=15, nb_pts=20,
+                 verbose=True):
+        """ Recognition of bundles
+
+        Extract bundles from a participants' tractograms using model bundles
+        segmented from a different subject or an atlas of bundles.
+        See [Garyfallidis17]_ for the details.
+
+        Parameters
+        ----------
+        streamlines : Streamlines
+            The tractogram in which you want to recognize bundles.
+        cluster_map : QB map
+            Provide existing clustering to start RB faster (default None).
+        clust_thr : float
+            Distance threshold in mm for clustering `streamlines`
+        nb_pts : int
+            Number of points per streamline (default 20)
+
+        Notes
+        -----
+        Make sure that before creating this class that the streamlines and
+        the model bundles are roughly in the same space.
+        Also default thresholds are assumed in RAS 1mm^3 space. You may
+        want to adjust those if your streamlines are not in world coordinates.
+
+        References
+        ----------
+        .. [Garyfallidis17] Garyfallidis et al. Recognition of white matter
+            bundles using local and global streamline-based registration and
+            clustering, Neuroimage, 2017.
+        """
+        self.clust_thr = clust_thr
+        self.streamlines = streamlines
+
+        self.nb_streamlines = len(self.streamlines)
+        self.verbose = verbose
+
+        self.start_thr = [40, 25, 20]
+
+        if cluster_map is None:
+            self._cluster_streamlines(clust_thr=clust_thr, nb_pts=nb_pts)
+        else:
+            if self.verbose:
+                t = time()
+
+            self.cluster_map = cluster_map
+            self.cluster_map.refdata = self.streamlines
+            self.centroids = self.cluster_map.centroids
+            self.nb_centroids = len(self.centroids)
+            self.indices = [cluster.indices for cluster in self.cluster_map]
+
+            if self.verbose:
+                print(' Streamlines have %d centroids'
+                      % (self.nb_centroids,))
+                print(' Total loading duration %0.3f sec. \n'
+                      % (time() - t,))
+
+    def _cluster_streamlines(self, clust_thr=15, nb_pts=20):
+
+        np.random.seed(42)
```

Comment:
Shouldn't this be a parameter? It would be nice if the user could simply choose a seed to have reproducable results. It's probably more complex than this simple line though?